src/Core/src/HotReload/HotReloadHelper.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Maui.Hosting;
+
+namespace Microsoft.Maui.HotReload
+{
+	public static class MauiHotReloadHelper
+	{
+		static IMauiServiceCollection? HandlerService;
+		//static IMauiHandlersServiceProvider? HandlerServiceProvider;
+		public static void Init(IMauiServiceCollection handlerService)
+		{
+			HandlerService = handlerService;
+			//HandlerServiceProvider = new MauiHandlersServiceProvider(handlerService);
+			IsEnabled = true;
+		}
+		public static void AddActiveView(IHotReloadableView view) => ActiveViews.Add(view);
+		public static void Reset()
+		{
+			replacedViews.Clear();
+		}
+		public static bool IsEnabled { get; set; } = Debugger.IsAttached;
+
+		public static void Register(IHotReloadableView view, params object[] parameters)
+		{
+			if (!IsEnabled)
+				return;
+			currentViews[view] = parameters;
+		}
+
+		public static void UnRegister(IHotReloadableView view)
+		{
+			if (!IsEnabled)
+				return;
+			currentViews.Remove(view);
+		}
+		public static bool IsReplacedView(IHotReloadableView view, IView newView)
+		{
+			if (!IsEnabled)
+				return false;
+			if (view == null || newView == null)
+				return false;
+
+			if (!replacedViews.TryGetValue(view.GetType().FullName, out var newViewType))
+				return false;
+			return newView.GetType() == newViewType;
+		}
+		public static IView GetReplacedView(IHotReloadableView view)
+		{
+			if (!IsEnabled)
+				return view;
+			if (!replacedViews.TryGetValue(view.GetType().FullName, out var newViewType))
+				return view;
+
+			currentViews.TryGetValue(view, out var parameters);
+			try
+			{
+				//TODO: Add in a way to use IoC and DI
+				var newView = (IView)(parameters?.Length > 0 ? Activator.CreateInstance(newViewType, args: parameters) : Activator.CreateInstance(newViewType));
```

Comment:
You can inject the IMauiServiceProvider and get it from there.. this could be done by init this helper on a configure delegate delegate.. 
Fonts is using a similar approach.